rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -562,18 +562,35 @@ void Player::prepare_publishers()
   reader_->set_filter(storage_filter);
 
   // Create /clock publisher
-  if (play_options_.clock_publish_frequency > 0.f) {
-    const auto publish_period = std::chrono::nanoseconds(
-      static_cast<uint64_t>(RCUTILS_S_TO_NS(1) / play_options_.clock_publish_frequency));
+  if (play_options_.clock_publish_frequency > 0.f || play_options_.clock_publish_on_topic_publish) {
```

Comment:
@kylemarcey `clock_publish_on_topic_publish` is mutual exclusive with `clock_trigger_topics` and `clock_publish_on_topic_publish` is false by default at least from Python binding API.
When `clock_trigger_topics` will be not empty it seems we will not create `clcok_publisher_` isn't it?

Reply:
@MichaelOrlov yes, good point. I now set `clock_publish_on_topic_publish` to true if either `--clock-topics-all` or `--clock-topics ...` are set so the publisher will be created in all cases